src/network/network_admin.cpp

```diff
@@ -333,11 +332,27 @@ NetworkRecvStatus ServerNetworkAdminSocketHandler::SendCompanyInfo(const Company
 	p->Send_uint8 (CeilDiv(c->months_of_bankruptcy, 3)); // send as quarters_of_bankruptcy
 
 	for (auto owner : c->share_owners) {
-		p->Send_uint8(owner);
+	  p->Send_uint8(owner);
+	}
+
+	std::vector<Company *> companies;
+	for(Company* company: Company::Iterate()){
+	  companies.emplace_back(company);
+	}
+
+	std::sort(companies.begin(), companies.end(), [](Company * const &c1, Company * const &c2){
+	  return c2->old_economy[0].performance_history < c1->old_economy[0].performance_history;
+	});
+
+	for(size_t t = 0; t < companies.size(); t++){
+	  companies[t]->rating = t+1;
 	}
 
+	p->Send_uint8 (c->rating);
```

Comment:
I have no clue what the impact is of just adding data to an admin-port packet. Do clients break? I think we have a version-number somewhere; it might be needed to bump that number. Maybe someone like @nielsmh knows?

Reply:
Each packet contains a 16 bit size field at the start, so clients that just ignore data at the end of the packet that they don't know about should keep working. I'm pretty sure clients are supposed to do that.